tests/e2e/isbservicerollout_test.go

```diff
@@ -0,0 +1,134 @@
+package e2e
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	numaflowv1 "github.com/numaproj/numaflow/pkg/apis/numaflow/v1alpha1"
+	apiv1 "github.com/numaproj/numaplane/pkg/apis/numaplane/v1alpha1"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+var _ = Describe("ISBService E2E", func() {
+	const (
+		namespace             = "numaplane-system"
+		isbServiceRolloutName = "test-isb-service-rollout"
+		timeout               = 30 * time.Second
+		duration              = time.Second
+		interval              = 250 * time.Millisecond
+	)
+	gvr := getGVRForISBServiceRollout()
+
+	BeforeEach(func() {
+		err := dynamicClient.Resource(gvr).Namespace(namespace).Delete(ctx, isbServiceRolloutName, metav1.DeleteOptions{})
+		if err != nil && !errors.IsNotFound(err) {
+			Expect(err).NotTo(HaveOccurred())
+		}
+	})
+
+	It("Should create, update, and delete a ISBService", func() {
+		// Creating ISB Service Rollout
+		By("Creating a new ISBServiceRollout")
+		isbServiceRollout := createISBServiceRolloutSpec(isbServiceRolloutName, namespace)
+		// fmt.Printf(isbServiceRollout)
+		err := createISBServiceRollout(ctx, isbServiceRollout)
+		Expect(err).NotTo(HaveOccurred())
+
+		Eventually(func() error {
+			_, err := dynamicClient.Resource(gvr).Namespace(namespace).Get(ctx, isbServiceRolloutName, metav1.GetOptions{})
+			return err
+		}, timeout, duration).Should(Succeed())
```

Comment:
Am I wrong that this is verifying that the ISBServiceRollout and not the ISBService exists? (since gvr is the Rollout)

Reply:
I switched the gvr from the rollout